doc/monads/laws.lean

```diff
@@ -0,0 +1,319 @@
+/-!
+# Monad Laws
+
+In the previous sections you learned how to use [Functors](functors.lean.md),
+[Applicatives](applicatives.lean.md), and [Monads](monads.lean.md), and you played with some useful
+instances including [Option](monads.lean.md), [IO](monads.lean.md), [Reader](readers.lean.md),
+[State](states.lean.md) and [Except](except.lean.md) and you learned about composition using [Monad
+Transformers](transformers.lean.md).
+
+So far, you've learned the concrete details you need in order to _use_ monads in your Lean programs.
+But there's still one more important concept you need if you want to _create_ new functors,
+applicatives and monads. Namely, the notion of _structural "laws"_ -- rules that these type
+classes should follow in order to meet other programmers' expectations about your code.
+
+## Life without Laws
+
+Remember Lean represents each of these abstract classes by a type class. Each of these type classes
```

Comment:
"abstract classes" => "abstract structures"

Reply:
Fixed in https://github.com/leanprover/lean4/pull/1505/commits/8dd24bc9dae40eb19b4625bf0b94cfb3b15a2f07